modules/nextflow/src/main/groovy/nextflow/k8s/client/K8sClient.groovy

```diff
@@ -387,7 +387,9 @@ class K8sClient {
 
             try {
                 return makeRequestCall( method, path, body )
-            } catch ( SocketException e ) {
+            } catch ( K8sResponseException | SocketException e ) {
+                if ( ( e instanceof K8sResponseException ) && ( e.response.code != 500 ))
```

Comment:
Being so, it would not be better to use `e.response.code >= 500` ? 

minor: the expression can be simplified as 

```
if ( e instanceof K8sResponseException && e.response.code != 500  ) 
```

Reply:
didn't you actually mean `code < 500` `->` throw the exception right now?

Reply:
My fault, I was misunderstanding the logic here. The rationale is to make `500` error code to behave like the `SocketException` and just rethrow in the other cases 👍

Reply:
Yes. Should I remove the parantheses and push?

Reply:
Yes, please